### PR TITLE
refactor: `tasks` module

### PR DESCRIPTION
**Pull request by phorward**
Refactoring the tasks module.

- Restructured `imports`
- Renamed preprocessJsonObject into _preprocess_json_object
- Renamed jsonDecodeObjectHook into _decode_object_hook
- Concise use of datetime-ISO-format
  - (addresses issue #486 as well!)
  - breaking change between viur-core 3.5 and viur-core 3.6 when applied to a project where tasks are still open

### CODE REVIEW

File: src/viur/core/tasks.py [hunk 1]
```diff
@@ -91,7 +93,7 @@ def jsonDecodeObjectHook(obj):
     taskClient = tasks_v2.CloudTasksClient()
 else:
     taskClient = tasks_v2.CloudTasksClient(
-        transport=CloudTasksGrpcTransport(channel=grpc.insecure_channel(os.getenv("TASKS_EMULATOR")))
+        transport=tasks_v2.services.cloud_tasks.transports(channel=grpc.insecure_channel(os.getenv("TASKS_EMULATOR")))
```
Comment (sveneberth) [inline src/viur/core/tasks.py:96]:
```suggestion
        transport=tasks_v2.services.cloud_tasks.transports.CloudTasksGrpcTransport(channel=grpc.insecure_channel(os.getenv("TASKS_EMULATOR")))
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: src/viur/core/tasks.py (hunk 1 above)
Comment (phorward) [reply src/viur/core/tasks.py:96]: Hoppela!

File: src/viur/core/tasks.py [hunk 2]
```diff
@@ -1,65 +1,67 @@
...
-    elif isinstance(o, (list, tuple, set)):
-        return [preprocessJsonObject(x) for x in o]
-    else:
-        return o
-
-
-def jsonDecodeObjectHook(obj):
+    if isinstance(obj, db.Key):
+        return {".__key__": db.encodeKey(obj)}
+    elif isinstance(obj, datetime):
```
Comment (sveneberth) [inline src/viur/core/tasks.py:29]:
```suggestion
    elif isinstance(obj, datetime.datetime):
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by sveneberth — APPROVED** [review]
With the fix https://github.com/viur-framework/viur-core/pull/1043 this seems to work

But please make the linter happy too :smile:

**Review by sveneberth — APPROVED** [review]

**Pull request merged**